http/nginx.conf

```diff
@@ -11,6 +11,22 @@ http {
     server_name nginx_gate;
 
     location / {
+      add_header 'Access-Control-Allow-Origin' '*';
+      add_header 'Access-Control-Allow-Credentials' 'true';
+      add_header 'Access-Control-Allow-Headers' 'Authorization,Accept,Origin,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Content-Range,Range';
+      add_header 'Access-Control-Allow-Methods' 'GET,POST,OPTIONS,PUT,DELETE,PATCH';
+
+      if ($request_method = 'OPTIONS') {
```

Comment:
Shouldn't the `if` be moved outside of `location` context? https://www.nginx.com/resources/wiki/start/topics/depth/ifisevil/

Reply:
We cannot move `if` out of `location`. It will be invalid nginx configuration. As I understood from [ifisevil](https://www.nginx.com/resources/wiki/start/topics/depth/ifisevil/) - our config quite valid (we use `return` and don't rely on directives out of `if`)

Reply:
If I understand correctly, using `return` doesn't make `if` safe, it is one of two things which is safe to use inside `if`

Reply:
> Use [try_files](https://nginx.org/en/docs/http/ngx_http_core_module.html#try_files) if it suits your needs. Use the “return …” or “rewrite … last” in other cases.

I think latter one is our case. In fact, this is purely test environment, we don't need to be perfect there on my opinion. @realloc can make this config better in future contributions :wink: